### DISCUSSION TITLE
which one to choose for using with Laravel?

### DISCUSSION THREAD

**Discussion by arlanram** [phpredis/phpredis#2354] [Q&A]
### I'm seeing this behaviour on
- OS: ubuntu 22.04
- Redis: 7.0.1
- PHP: 7.4fpm-alpine-3.16
- phpredis: latest

please give an advice

1. Redis::SERIALIZER_IGBINARY vs Redis::SERIALIZER_MSGPACK 
2. COMPRESSION_LZF vs COMPRESSION_LZ4

which one is better to use, less consumtion of space and memory and also secure in production.

**Comment by arlanram** [comment]
@yatsukhnenko please help

**Comment by yatsukhnenko** [comment]
@arlanram unfortunately I don't have any benchmark results for specific framework to suggest.

**Reply by arlanram** [reply]
but in general? which ones is better? couldn't find anything on the internet

**Reply by hotrush** [reply]
Both igbinary and msgpack are better then json or native serialize because of faster serialization and lower memory usage. IMO, igbinary is better for redis because it provides faster unserializes on reads. We use igbinary for years in production and used it even with memcached many years ago, no any issues.
